packages/nx/src/generators/tree.ts

```diff
@@ -357,20 +361,16 @@ export class FsTree implements Tree {
 
   private assertUnlocked() {
     if (this.locked) {
-      // TODO (v17): Remove condition
-      if (gt(nxVersion, '17.0.0')) {
-        throw new Error(
-          'The tree has already been committed to disk. It can no longer be modified. Do not modify the tree during a GeneratorCallback and ensure that Promises have resolved before the generator returns or resolves.'
-        );
-      } else {
-        output.warn({
-          title: 'Tree modified after commit to disk.',
-          bodyLines: [
-            'The tree has already been committed to disk. It can no longer be modified. Do not modify the tree during a GeneratorCallback and ensure that Promises have resolved before the generator returns or resolves.',
-            `This will be an error in version 16. Please open an issue on the Nx repo if experiencing this with a first-party plugin, or the plugin's repo if using a community plugin.`,
-          ],
-        });
-      }
+      output.error({
+        title: `File changes have already been written to disk. Further changes were attempted ${
+          this.logOperationId ? `while running ${this.logOperationId}.` : '.'
+        }`,
+        bodyLines: [
+          'It can no longer be modified. This commonly happens when a generator attempts to make further ',
```

Comment:
Now there's no previous mention of the tree. I think the `It` here lacks context about what it's referring to.
```suggestion
          'The tree can no longer be modified. This commonly happens when a generator attempts to make further ',
```